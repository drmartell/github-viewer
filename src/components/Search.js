import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { fetchUser, fetchRepos, searching } from '../actions/actions';
import UserSection from './UserSection';
import ReposSection from './ReposSection';

const Search = () => {
  const dispatch = useDispatch();
  const [searchString, setSearchString] = useState();
  
  const handleSubmit = event => {
    event.preventDefault();
    dispatch(searching());
    dispatch(fetchUser(searchString));
    dispatch(fetchRepos(searchString));
  };

  return <>
    <form onSubmit={handleSubmit}>
      <input type="text" name="search" onChange={({ target: { value } }) => setSearchString(value) }/>
      <button>Search</button>
    </form>

    <UserSection />
    <ReposSection />
  </>;
};

export default Search;
