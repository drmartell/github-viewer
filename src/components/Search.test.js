import React from 'react';
import { mount } from 'enzyme';
import Search from './Search';
import { Provider } from 'react-redux';
import store from '../store';

describe('Search component', () => {
  it('renders Search', () => {
    const wrapper = mount(
      <Provider store={store}>
        <Search />
      </Provider>);
    expect(wrapper).toMatchSnapshot();
  });
});
