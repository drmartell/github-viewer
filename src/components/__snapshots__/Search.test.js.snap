// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders Search 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Search>
    <form
      onSubmit={[Function]}
    >
      <input
        name="search"
        onChange={[Function]}
        type="text"
      />
      <button>
        Search
      </button>
    </form>
    <UserSection />
    <ReposSection>
      <Loading>
        <img
          src="https://mir-s3-cdn-cf.behance.net/project_modules/disp/2a6dae53616869.593a85ce553c4.gif"
          style={
            Object {
              "display": "block",
              "height": "auto",
              "margin": "auto",
              "width": "10vw",
            }
          }
        />
      </Loading>
    </ReposSection>
  </Search>
</Provider>
`;
